xarray/core/utils.py

```diff
@@ -936,3 +936,21 @@ def iterate_nested(nested_list):
             yield from iterate_nested(item)
         else:
             yield item
+
+
+def contains_only_dask_or_numpy(obj) -> bool:
```

Comment:
This is used to send objects with only dask or numpy data to `dask_groupby`. otherwise we use the legacy code path.